core/dbt/contracts/graph/nodes.py

```diff
@@ -684,6 +686,19 @@ def same_contents(self, other) -> bool:
     def test_node_type(self):
         return "generic"
 
+    def write_node(self, target_path: str, subdirectory: str, payload: str):
+        # If this test came from yaml frontmatter, it will try to write to a directory
+        # with the same name as the model file, which won't work. So change the
+        # directory to model file name + .yaml.
+        file_path = self.original_file_path
+        if file_path.endswith(".sql"):
```

Comment:
- Need to check for `".py"` here as well, for Python models
- Maybe it should just write to the same directory that the model is defined in? Need to think a bit more about this. I don't think it's a high-stakes decision one way or the other

Reply:
I *think* that if we put it in the same directory that the model is in we could have collisions with other tests from other models in the same directory.